m4/libmesh_metaphysicl.m4

```diff
@@ -58,7 +58,28 @@ AC_DEFUN([CONFIGURE_METAPHYSICL],
           METAPHYSICL_INCLUDE="-I\$(top_srcdir)/contrib/metaphysicl/src/numerics/include -I\$(top_srcdir)/contrib/metaphysicl/src/core/include -I\$(top_srcdir)/contrib/metaphysicl/src/utilities/include"
           AC_DEFINE(HAVE_METAPHYSICL, 1, [Flag indicating whether the library will be compiled with MetaPhysicL support])
           AC_MSG_RESULT(<<< Configuring library with MetaPhysicL support >>>)
-          AX_SUBDIRS_CONFIGURE([contrib/metaphysicl],[--with-cxx-std=20$acsm_cxx_version])
+
+          dnl MetaPhysicL needs to know which TIMPI library to link to
+          dnl for "make check"
+          my_method=dbg
+          AS_IF([test "x$build_devel" = xyes],
+                [my_method=devel])
+          AS_IF([test "x$build_prof" = xyes],
+                [my_method=prof])
+          AS_IF([test "x$build_oprof" = xyes],
+                [my_method=oprof])
+          AS_IF([test "x$build_opt" = xyes],
+                [my_method=opt])
+
+          dnl Autoconf doesn't define $abs_top_srcdir at this point;
+          dnl here's a trick from GraphicsMagick:
+          my_top_srcdir="$(cd $srcdir && pwd)"
+
+          dnl FIXME: setting TIMPI_DIR, even to something invalid, is
+          dnl currently the best way to get MetaPhysicL to recognize
+          dnl that we really want TIMPI.
+
+          AX_SUBDIRS_CONFIGURE([contrib/metaphysicl],[[--with-cxx-std=20$acsm_cxx_version],[CPPFLAGS=-I$my_top_srcdir/contrib/timpi/src/algorithms/include/ -I$my_top_srcdir/contrib/timpi/src/parallel/include/ -I$my_top_srcdir/contrib/timpi/src/utilities/include/ -I$ac_abs_top_builddir/contrib/timpi/src/utilities/include/ $CPPFLAGS],[LDFLAGS=-L$ac_abs_top_builddir/contrib/timpi/src/ $LDFLAGS],[TIMPI_DIR=$my_top_srcdir/contrib/timpi/],[--with-timpi-method=$my_method]])
```

Comment:
I thought that I'd set things up such that `TIMPI_DIR` needed to point to a directory whose sub-directories were `bin, include, lib` but here it looks like we are now pointing to a source directory. What else has changed such that this works? I see all the proper flags getting set here so now I'm wondering if `TIMPI_DIR` is even relevant at all?

Reply:
A mass of manually-set CPPFLAGS and LDFLAGS changed such that this works.  TIMPI_DIR is only relevant because its existence is at the moment what convinces timpi.m4 that we want to try to configure with TIMPI, so that it has the chance to discover that the manual flags work.  In hindsight the m4 idioms I was using a decade ago when some of these m4 files were originally written were not the appropriate idioms to import into libMesh.  In the long run I want to rewrite probably 75% of timpi.m4 and 25% of timpi_control_args.m4 but for this PR I was shooting for the minimal changeset that would get CI working again.